src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -1036,15 +1062,19 @@ export class TerminalInstance implements ITerminalInstance {
 	}
 
 	private _attachPressAnyKeyToCloseListener() {
-		this._processManager!.addDisposable(dom.addDisposableListener(this._xterm.textarea, 'keypress', (event: KeyboardEvent) => {
+		const keyPressListener = dom.addDisposableListener(this._xterm.textarea, 'keypress', (event: KeyboardEvent) => {
```

Comment:
Why is this change needed?

Reply:
Because in the case of a task execution, there is no process manager but we still want terminal re-use, i.e. the text that says "Terminal will be reused, press any key to exit..."

Reply:
We still need to add this to this._processManager.addDisposable if we have a _processManager (and TerminalInstance.addDisposable if not), otherwise the event listener will still live for example when waitOnExit happens and then dispose is clicked. I suspect this might cause a memory leak otherwise.

Reply:
Nice catch!
Addressed in a non-process specific manner.